R/addDissimilarity.R

```diff
@@ -0,0 +1,312 @@
+#' Calculate dissimilarities
+#'
+#' These functions are designed to calculate dissimilarities on data stored 
+#' within a 
+#' \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#'  object. For overlap, Unifrac, and Jensen-Shannon Divergence (JSD) 
+#'  dissimilarities, the functions use mia internal functions, while for other 
+#'  types of dissimilarities, they rely on \code{\link[vegan:vegdist]{vegdist}}
+#'  by default.
+#'
+#' @param x a \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#'   object.
+#'
+#' @param method \code{Character scalar}. Specifies which distance to calculate.
+#' 
+#' @param name \code{Character scalar}. The name to be used to store the result 
+#'  in the reducedDims of the output. (Default: \code{method})
+#'
+#' @param assay.type \code{Character scalar}. Specifies which assay to use for 
+#'   calculation. (Default: \code{"counts"})
+#'
+#' @param assay_name Deprecated. Use \code{assay.type} instead.
+#'   
+#' @param exprs_values Deprecated. Use \code{assay.type} instead.
+#'   
+#' @param transposed \code{Logical scalar}. Specifies if x is transposed with
+#'   cells in rows. (Default: \code{FALSE})
+#'
+#' @param tree.name \code{Character scalar}. Specifies the name of the tree
+#'   used in calculation. (Default: \code{"phylo"})
+#' 
+#' @param tree_name Deprecated. Use \code{tree.name} instead.
+#'
+#' @param ... other arguments passed onto \code{\link[vegan:vegdist]{vegdist}}, 
+#'  or the following arguments passed onto mia internal functions for overlap,
+#'  unifrac and JSD dissimilarities:
+#' \itemize{
+#'   \item \code{weighted}: (Unifrac) \code{Logical scalar}. Should use
+#'   weighted-Unifrac calculation? 
+#'   Weighted-Unifrac takes into account the relative abundance of
+#'   species/taxa shared between samples, whereas unweighted-Unifrac only
+#'   considers presence/absence. Default is \code{FALSE}, meaning the
+#'   unweighted-Unifrac distance is calculated for all pairs of samples.
+#'   (Default: \code{FALSE})
+#'   
+#'   \item \code{chunkSize}: (JSD) \code{Integer scalar}. Defines the size of
+#'   data  send to the individual worker. Only has an effect, if \code{BPPARAM} 
+#'   defines more than one worker. (Default: \code{nrow(x)})
+#'   
+#'   \item \code{BPPARAM}: (JSD)
+#'   \code{\link[BiocParallel:BiocParallelParam-class]{BiocParallelParam} object}.
+#'   Specifies whether the calculation should be parallelized.
+#'   
+#'   \item \code{detection}: (overlap) \code{Numeric scalar}.
+#'   Defines detection threshold for absence/presence of features. Feature that 
+#'   has abundance under threshold in either of samples, will be discarded when 
+#'   evaluating overlap between samples. (Default: \code{0}) 
+#' }
+#'
+#' @return 
+#' \code{getDissimilarity} returns a sample-by-sample distance matrix, suitable 
```

Comment:
distance matrix -> dissimilarity matrix? (also check elsewhere, let's be consistent)